lib/AST/ASTPrinter.cpp

```diff
@@ -4127,6 +4165,8 @@ void PrintAST::visitFuncDecl(FuncDecl *decl) {
       }
 
       PrintWithOpaqueResultTypeKeywordRAII x(Options);
+      PrintForFunctionResultRAII assumingFunctionResult(Options);
```

Comment:
Is this not relevant when printing closure argument types as well?

Reply:
Closure arguments were fine... closure results were not. I added fixes + tests to validate the behavior. So it works now.